### DISCUSSION TITLE
Solidity Language Support

### DISCUSSION THREAD

**Discussion by Gostycw** [zed-industries/zed#6517] [Feature Requests]
When will Zed support the Solidity language? This is important for blockchain developers.

**Comment by Evalir** [comment]
Bump on this! It's the only thing I need to switch full time to the editor.

**Comment by koloz193** [comment]
would love this as well! I've been using it without any syntax highlighting and even tried to force it to use js highlights but its not the same

**Comment by JosephTLyons** [comment]
Hey all, could everyone who put an upvote-like emoji here go put a 👍 emoji on this issue? 

- https://github.com/zed-industries/community/issues/1292

We track interest in issues using a script that counts the 👍 on issues.  The upvotes here, unfortunately, won't be counted.

**Comment by MDSL21** [comment]
Great thing! Agree!

**Comment by SomeoneToIgnore** [comment]
Issue: https://github.com/zed-industries/zed/issues/4872
PR: https://github.com/zed-industries/zed/pull/7152

**Comment by wiasliaw** [comment]
solc provides lsp service following Language Server Protocol. I think the config for neovim can convert into config for zed.

> vim / neovim: You need the coc plugin and add the following to the coc-settings.json file:
```
{
    // ...
    "languageserver": {
        // ...
        "solidity": {
            "command": "/path/to/solc",
            "args": [ "--lsp" ],
            "rootPatterns": [".git/"],
            "filetypes": ["solidity"]
        }
    }
}
```

reference: https://soliditylang.org/blog/2021/12/20/solidity-0.8.11-release-announcement/
